### PR TITLE
add callback to websocket-setup-env

### PR DESCRIPTION
**Pull request by piranha**
As mentioned in #63, I need to be able to determine a port where weasel is listening in order to connect there. What do you think of this approach?

### CODE REVIEW

**Comment by tomjakubowski** [comment]
Nice! I didn't know http-kit supported running on an arbitrary available port. That's good to know about.

LGTM, but I'd prefer to name it something better than init. If we (or someone) can't come up a better one I'll happily merge though.

File: src/clj/weasel/repl/websocket.clj [hunk 1]
```diff
@@ -64,10 +64,12 @@
     (fn [data] (process-message this (read-string data)))
     :ip (:ip this)
     :port (:port this))
-  (let [{:keys [ip port]} this]
-    (println (str "<< started Weasel server on ws://" ip ":" port " >>"))
+  (let [{:keys [ip port init]} this]
```
Comment (tomjakubowski) [inline src/clj/weasel/repl/websocket.clj]: you could remove port from the `:keys` map, right?

**Comment by piranha** [comment]
Yeah, `init` was a fast one and I don't like it either. It was a `callback` initially, which also doesn't convey a lot of information. How about `connection-hook`?

**Comment by tomjakubowski** [comment]
Since the hook is called before the connection, what do you think of `pre-connect`? It has a nice symmetry with `on-connect` et al. in the client.

**Comment by piranha** [comment]
I like that! Changed. :)

**Comment by tomjakubowski** [comment]
Thanks so much!

**Comment by piranha** [comment]
Tom, can you please make a release? :-) I really want to stop supporting my own forks of weasel and boot-cljs-repl, and if you release weasel, boot-cljs-repl can be fixed for everyone. :)

**Comment by martinklepsch** [comment]
That sounds great, fixing `boot-cljs-repl` as well has been long outstanding. :)

**Comment by tomjakubowski** [comment]
Will release 0.7.0 after work today.

**Comment by tomjakubowski** [comment]
0.7.0 is released. Thanks for reminding me.

**Comment by piranha** [comment]
Thanks! :)

**Comment by defclass** [comment]
Hi piranha,  I have one strange problem, when client connected ,the console will be blocked .  The cursor can move , but can never evaluate form any more.

```
user=> (require 'weasel.repl.websocket)
nil
user=> (cemerick.piggieback/cljs-repl
  #_=>         (weasel.repl.websocket/repl-env :ip "0.0.0.0" :port 9001))
<< started Weasel server on ws://0.0.0.0:9001 >>
<< waiting for client to connect ...  connected! >>

```
version is below
```
[weasel "0.7.0" :exclusions [org.clojure/clojurescript]]
```

**Pull request merged**